x-pack/solutions/search/plugins/enterprise_search/server/deprecations/routes.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema } from '@kbn/config-schema';
+
+import { deleteConnectorById, putUpdateNative } from '@kbn/search-connectors';
+
+import { RouteDependencies } from '../plugin';
+
+import { elasticsearchErrorHandler } from '../utils/elasticsearch_error_handler';
+
+export function registerDeprecationRoutes({ router, log }: RouteDependencies) {
+  router.post(
+    {
```

Comment:
who should have access to this route? it being destructive I think we likely want to tie it to some role/feature.

Reply:
Since your using `client.asCurrentUser` then we are _probably_ fine here. But I asked since we need to be more aware of when we should be setting `security` on routes and ensure API routes are restricted based on RBAC.

[playground example](https://github.com/elastic/kibana/blob/main/x-pack/solutions/search/plugins/search_playground/server/routes.ts#L60-L64)